src/gui/qt/gui.cpp

```diff
@@ -27,197 +27,286 @@ const int yOffset = -380;
-    QMainWindow mainWindow;
+int main(int argc, char *argv[]) {
+  QApplication app(argc, argv);
+  QMainWindow mainWindow;
 
-    // This needs to be insatlled to everything
-    filter = new KeyboardEventFilter;
+  // This needs to be insatlled to everything
+  filter = new KeyboardEventFilter;
 
-    mainWindow.installEventFilter(filter);
+  mainWindow.installEventFilter(filter);
 
-    // Create a central widget to hold the canvas and buttons
-    QWidget* centralWidget = new QWidget(&mainWindow);
-    centralWidget->installEventFilter(filter);
-    mainWindow.setCentralWidget(centralWidget);
+  // Create a central widget to hold the canvas and buttons
+  QWidget *centralWidget = new QWidget(&mainWindow);
+  centralWidget->installEventFilter(filter);
+  mainWindow.setCentralWidget(centralWidget);
 
-    // Create a vertical layout to arrange the canvas and buttons
-    QVBoxLayout* layout = new QVBoxLayout(centralWidget);
-    layout->installEventFilter(filter);
+  // Create a vertical layout to arrange the canvas and buttons
+  QVBoxLayout *layout = new QVBoxLayout(centralWidget);
+  layout->installEventFilter(filter);
 
-    // Create the canvas view
-    canvasView = new CanvasView();
-    canvasView->setFixedSize(396, 224);
-    canvasView->installEventFilter(filter);
-    layout->addWidget(canvasView, 0, Qt::AlignTop);
+  // Create the canvas view
+  canvasView = new CanvasView();
+  canvasView->setFixedSize(396, 224);
+  canvasView->installEventFilter(filter);
+  layout->addWidget(canvasView, 0, Qt::AlignTop);
 
-    ButtonContainer container;
-    // Make it capture keyboard events
-    container.setFocusPolicy(Qt::StrongFocus);
-    container.installEventFilter(filter);
-    layout->addWidget(&container, 0, Qt::AlignTop);
+  ButtonContainer container;
+  // Make it capture keyboard events
+  container.setFocusPolicy(Qt::StrongFocus);
+  container.installEventFilter(filter);
+  layout->addWidget(&container, 0, Qt::AlignTop);
 
-    // Add a stretch at the end to take any available extra space
-    layout->addStretch();
+  // Add a stretch at the end to take any available extra space
+  layout->addStretch();
 
-    // Create a menu bar and menus
-    // QMenuBar* menuBar = mainWindow.menuBar();
-    // QMenu* fileMenu = menuBar->addMenu("File");
+  // Create a menu bar and menus
+  // QMenuBar* menuBar = mainWindow.menuBar();
+  // QMenu* fileMenu = menuBar->addMenu("File");
 
-    // Add actions to the menus if needed
-    // QAction* openAction = fileMenu->addAction("Open .g3a file");
+  // Add actions to the menus if needed
+  // QAction* openAction = fileMenu->addAction("Open .g3a file");
 
-    mainWindow.show();
+  mainWindow.show();
 
-    const char* filePath;
-    if (argc < 2) {
-        QString qFilePath = QFileDialog::getOpenFileName(&mainWindow, "Open g3a add-in", "", "Add-in files (*.g3a)");
-        filePath = qFilePath.toUtf8();
-    } else {
-        filePath = argv[1];
-    }
+  char filePath[1024];
```

Comment:
This buffer could overflow if the file name is too long. It won't matter most of the time but it would still be better to avoid it. Three possible solutions: Move the declaration of the QString backwards so it's still valid at the time when it's used, use `strdup` to allocate the right amount of memory, or increase the size of the array to 4096 (what Google says is the max file path length on Linux) and use `strncpy` just in case it's somehow longer anyway so it gets cut off instead of overflowing.

>  Now the fileopen dialog does not crash anymore when not using the argv[1] parameter to assign the name of the addin.

I'm a bit confused by what you mean here. If the name is assigned then the dialog code should never get run at all, shouldn't it?